lib/daru/vector.rb

```diff
@@ -352,7 +352,14 @@ def == other
         if other.is_a?(Daru::Vector)
           mod.apply_vector_operator operator, self, other
         else
-          mod.apply_scalar_operator operator, @data,other
+          missing_undefined_indexes = nil_positions + nan_positions
+          if operator != :== && operator != :!= && missing_undefined_indexes.size
+            data_dup = @data.dup
+            missing_undefined_indexes.each { |i| data_dup.delete_at(i) }
+            mod.apply_scalar_operator operator, data_dup, other
+          else
+            mod.apply_scalar_operator operator, data, other
+          end
```

Comment:
Thanks for review and telling me `Float::NAN == Float::NAN # => false`.This must be done in `apply_scalar_operator `. I will try it.